traffic_ops/traffic_ops_golang/cachesstats/cachesstats_test.go

```diff
@@ -0,0 +1,128 @@
+package cachesstats
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import (
+	"testing"
+	"time"
+
+	"github.com/apache/trafficcontrol/lib/go-tc"
+	"github.com/apache/trafficcontrol/lib/go-util"
+)
+
+func TestAddStatsInvalidString(t *testing.T) {
+	cacheData := make([]CacheData, 0)
+	cacheServerStats := make(map[string]tc.ServerStats, 0)
+	stats := make(map[string][]tc.ResultStatVal, 0)
+	resultStatVals := make([]tc.ResultStatVal, 0)
+	resultStatVal := tc.ResultStatVal{
+		Span: 0,
+		Time: time.Now(),
+		Val:  "invalid bandwidth",
```

Comment:
Do you happen to have an example of the values that we were actually receiving that was causing this bug? I see in the panic that it considers the interface a string type. Is that what happens when we get an empty JSON object?

Reply:
I do not have it handy, but I remember that this came up after 5.0 got deployed. I can dig in the TM logs and find another example if you'd like. I did adda test for empty json objects though.

Reply:
Yeah, in the bug report is says `panic: (err: interface conversion: interface {} is string, not uint64)`, just was caught offguard by the empty interface being typed as a string. I think this is fine and can be resolved.